### ISSUE TITLE
Version 4.2.1 Removed Support for Ruby 2.1 & 2.2

### ISSUE THREAD

**Issue by mike-stewart** [rocketjob/semantic_logger#104]
Was this intentional? 

The Changelog does not mention it, and it's quite unusual for a patch version to cause breaking changes. The README also still says that Ruby 2.1+ is supported.

Changed here: https://github.com/rocketjob/semantic_logger/commit/b1f09fc77333386ae8d59f4e61867e157972ae9c#diff-2737c44c682ec74fe0955bbf1c92e631R20

**Comment by reidmorrison** [comment]
If you are running Ruby 2.1 or 2.2, bundler will not see the newer gem version on older Ruby versions, so it will not impact anyone running `bundle update` on the older Ruby versions.
Ruby 2.1 and 2.2 were causing a security compliance violation. Recent security patches were not applied to those versions and they should therefore not be used.
RuboCop reported these discrepancies in the gemspec, so it was fixed as part of the maintenance/security patch. The readme can also be updated accordingly.

**Comment by mike-stewart** [comment]
Understood. I only flagged it because I thought it might have been accidental to introduce a breaking change in a patch version rather than a major or minor release. Wasn't sure if this project followed semver.

Bundler does break using bundler 1.16.1 (latest):
```
bundle update rails_semantic_logger
...
Fetching semantic_logger 4.2.1 (was 4.2.0)
Installing semantic_logger 4.2.1 (was 4.2.0)
Gem::InstallError: semantic_logger requires Ruby version >= 2.3.
An error occurred while installing semantic_logger (4.2.1), and Bundler cannot continue.
Make sure that `gem install semantic_logger -v '4.2.1'` succeeds before bundling.

In Gemfile:
  rails_semantic_logger was resolved to 4.2.1, which depends on
    semantic_logger
```

For those of us on older versions of ruby, will this gem no longer work? Or was this just to satisfy the rubocop check? If the latter, I'll fork it as needed.

Thanks for the useful gem!

**Comment by reidmorrison** [comment]
RuboCop has automated tools to upgrade the code to use some of the newer language features, so I don't think it will work 100% on Ruby 2.2.

Semantic Logger is tested against all of the following Ruby versions, so over time we have to stop supporting old Ruby versions:
  - 2.3.7
  - 2.4.4
  - 2.5.1
  - 2.6.0-preview1
  - jruby-9.1.17.0

Hopefully you are able to upgrade to a newer Ruby version soon.
